### PR TITLE
Fixes #7477 - Add PSNativeCommandArgumentPassing

### LINKED ISSUE #7477: Document new experiemental feature  PSNativeCommandArgumentPassing
# Summary of the new document or enhancement

- List of articles that need to be updated: https://docs.microsoft.com/powershell/scripting/learn/experimental-features
- New articles that need to be created:
- Link(s) to related code PR(s) in the PowerShell/PowerShell repo: PowerShell/PowerShell#14692
- Link(s) to related issue(s) in the PowerShell/PowerShell repo:

## Description of what changed

Document new Experimental feature -  PSNativeCommandArgumentPassing: Use ArgumentList for native executable invocation (breaking change)

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Fixes [AB#1836620](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1836620) - Fixes #7477 - Add PSNativeCommandArgumentPassing

Select the area of the Table of Contents containing the documents being changed.

**Conceptual content**
- [ ] Overview and Install
- [x] Learning PowerShell
  - [ ] PowerShell 101
  - [ ] Deep dives
  - [ ] Remoting
- [ ] Release notes (What's New)
- [ ] Windows PowerShell
  - WMF, ISE, release notes, etc.
- [ ] DSC articles
- [ ] Community resources
- [ ] Sample scripts
- [ ] Gallery articles
- [ ] Scripting and development
  - [ ] Legacy SDK

**Cmdlet reference & about_ topics**
- [ ] Preview content
- [ ] Version 7.1 content
- [ ] Version 7.0 content
- [ ] Version 5.1 content

## PR Checklist

- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] PR is targeted at the _staging_ branch
- [x] All relevant versions updated
- [x] Includes content related to issues and PRs - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**
  - If the PR is work in progress, please add the prefix `WIP:` or `[WIP]` to the beginning of the
    title and remove the prefix when the PR is ready.

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

File: reference/docs-conceptual/learn/experimental-features.md [hunk 1]
```diff
@@ -396,3 +397,89 @@ The experimental feature includes a new cmdlet,
...
+When this experimental feature is enabled PowerShell uses the `ArgumentList` property of the
+`StartProcessInfo` object rather than our current mechanism of reconstructing a string when invoking
+a native executable.
+
+This feature adds a new automatic variable `$PSNativeCommandArgumentPassing` that allows you to
+select the behavior at runtime. The valid values are `Legacy` and `Standard`. `Legacy` is the
+historic behavior. The default when the experimental feature is enabled is the new `Standard`
+behavior.
+
+The new behavior _is a breaking change_ from current behavior. This may break scripts and
```
Comment (chasewilson) [inline reference/docs-conceptual/learn/experimental-features.md:412]: Should this be a warning?

**Review by chasewilson — APPROVED** [review]

File: reference/docs-conceptual/learn/experimental-features.md (hunk 1 above)
Comment (sdwheeler) [reply reference/docs-conceptual/learn/experimental-features.md:412]: No, not during the experimental phase.

Comment (sdwheeler) [reply reference/docs-conceptual/learn/experimental-features.md]: Changed my mind. Added CAUTION.

**Review by chasewilson — APPROVED** [review]

File: reference/docs-conceptual/learn/experimental-features.md [hunk 2]
```diff
@@ -396,3 +397,90 @@ The experimental feature includes a new cmdlet,
...
+select the behavior at runtime. The valid values are `Legacy` and `Standard`. `Legacy` is the
+historic behavior. The default when the experimental feature is enabled is the new `Standard`
+behavior.
+
+> [!CAUTION]
+> The new behavior is a **breaking change** from current behavior. This may break scripts and
+> automation that work-around the various issues when invoking native applications. Historically,
+> quotes must be escaped and it is not possible to provide empty arguments to a native application.
+
+New behaviors made available by this change
```
Comment (chasewilson) [inline reference/docs-conceptual/learn/experimental-features.md:417]:
```suggestion
New behaviors made available by this change:
```

File: reference/docs-conceptual/learn/experimental-features.md [hunk 3]
```diff
@@ -396,3 +397,90 @@ The experimental feature includes a new cmdlet,
...
+
+- Literal or expandable strings with embedded quotes the quotes are now preserved
```
Comment (chasewilson) [inline reference/docs-conceptual/learn/experimental-features.md:419]:
```suggestion
- Literal or expandable strings with embedded quotes the quotes are now preserved:
```

**Review by chasewilson — APPROVED** [review]

**Pull request merged**